### PR TITLE
ENT-248: Completion installation

### PR DESCRIPTION
**Pull request by toothbrush**
# Description

We're trying to make the onboarding as smooth as possible.

* Optimise for a user that has never seen us before - guide them through setting up PATH.
* If all goes well, call `curl-bash-post-install` action, which will attempt to set up shell completions

There are a lot of trade offs here and i don't know if we can achieve perfection.  I couldn't think of a nice, robust way to ensure the PATH stuff, while remaining friendly and "one-shot" from the user's perspective.  And between shell completions and being on `PATH`, clearly `PATH` is highest priority.

So, if the user **did not** have a `~/.local/bin` on their PATH, we make sure that gets done.  But, in the happy path case, we set up completions, too.  On the bright side, if they later want to upgrade (which is just `curl | bash` again) we'll automatically guide them through the completions setup at that point.

I'm open to feedback on this one but it's a tough set of tradeoffs.  I think this is acceptable as is, and we can circle back if not.

### Trivia

Along the way i discovered:

* I would like to try and detect Fish and offer installation - separate PR
* I have a simple approach up my sleeve for allowing us to clean up completion config on uninstall – that'll also be a separate PR.
* Need to fix the issue actually reported as ENT-248 but that's easy.
* ~Worse: if we're telling users to `curl|bash`, the install script will always be running Bash.  We should probably just ask users which shell they'd like to install completions for?  Hmmmm...~ OK, it turns out `$SHELL` is "special".  It's set to your login shell's path, and if you invoke some other shell (say you're running Zsh and execute `bash`) it **won't** be overridden.  So it's an ok approach to use as we do.

---

> [!NOTE]
> **Medium Risk**
> Touches the installation/onboarding path and modifies how the binary is discovered on `PATH`, which can cause user-visible install failures or confusing behavior if edge cases are missed; changes are localized and covered by a small unit test for shell target selection.
> 
> **Overview**
> Improves onboarding for the `curl | bash` installer by adding PATH guidance for first-time installs and detecting PATH conflicts, then running a new hidden CLI post-install command to optionally set up shell completions.
> 
> Moves shell completion setup out of `entire enable` (removes `--setup-shell` and related non-interactive logic) and refactors completion targeting into `shellCompletionTarget()` with better bash rc-file selection (`.bash_profile` vs `.bashrc`) plus tests.
> 
> Updates release packaging to generate and ship `completions/*` in archives, and configures Homebrew cask to install bash/zsh/fish completions.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c269cb24f6bfc2bbf62cacf9b6b1d70f596599d3. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by toothbrush** [comment]
This is related to ENT-248 but isn't a fix, yet.  There will be followups.

File: cmd/entire/cli/setup.go [hunk 1]
```diff
@@ -678,46 +654,66 @@ func newSetupGitHookCmd() *cobra.Command {
...
-		completionLine = "source <(entire completion zsh)"
+		return "Zsh",
+			filepath.Join(home, ".zshrc"),
+			"autoload -Uz compinit && compinit && source <(entire completion zsh)"
 	case strings.Contains(shell, "bash"):
-		rcFile = filepath.Join(home, ".bashrc")
-		completionLine = "source <(entire completion bash)"
+		return "Bash",
+			filepath.Join(home, ".bashrc"),
+			"source <(entire completion bash)"
```
[the remark this answers is not in the document]
Comment (toothbrush) [reply cmd/entire/cli/setup.go]: I think it's acceptable to assume that since we're warning the user, and we're telling them which rc file we've changed, they'll figure it out.

We could also make that situation more obvious by warning about PATH first, then running post-install.  I don't think we're ever going to be able to address every single edge-case.

File: scripts/install.sh [hunk 2]
```diff
@@ -220,12 +214,13 @@ main() {
...
     if "$install_path" version &> /dev/null; then
         success "Entire CLI installed successfully!"
-        echo ""
-        echo "Run 'entire --help' to get started."
     else
         error "Installation completed but the binary failed to execute. Please check the installation."
     fi
 
+    info "Running post-install actions..."
+    "$install_path" curl-bash-post-install
```
[the remark this answers is not in the document]
Comment (toothbrush) [reply scripts/install.sh]: I tested this locally and had no trouble.  I'll shelve this until we see issues.

File: cmd/entire/cli/setup.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (toothbrush) [reply cmd/entire/cli/setup.go]: To be extra safe, i've added a guard in https://github.com/entireio/cli/pull/180/commits/b7d1ba0418e8d07a5ad42b2cd88f821a02a1b3a8

File: cmd/entire/cli/setup.go [hunk 3]
```diff
@@ -678,46 +654,66 @@ func newSetupGitHookCmd() *cobra.Command {
...
 	default:
-		return nil // Unsupported shell, skip silently
+		return "", "", ""
+	}
+}
```
[the remark this answers is not in the document]
Comment (toothbrush) [reply cmd/entire/cli/setup.go:705]: I think the logic is simple enough that it's fine.

File: cmd/entire/cli/setup.go [hunk 4]
```diff
@@ -678,46 +654,66 @@ func newSetupGitHookCmd() *cobra.Command {
...
 	switch {
 	case strings.Contains(shell, "zsh"):
-		rcFile = filepath.Join(home, ".zshrc")
-		completionLine = "source <(entire completion zsh)"
+		return "Zsh",
+			filepath.Join(home, ".zshrc"),
+			"source <(entire completion zsh)"
 	case strings.Contains(shell, "bash"):
-		rcFile = filepath.Join(home, ".bashrc")
-		completionLine = "source <(entire completion bash)"
```
[the remark this answers is not in the document]
Comment (toothbrush) [reply cmd/entire/cli/setup.go:705]: Addressed in https://github.com/entireio/cli/pull/180/commits/e0324212b66bf445a941ab70c83f31376e5a98d5

File: scripts/install.sh [hunk 5]
```diff
@@ -240,6 +232,7 @@ main() {
         echo -e "${YELLOW}!${NC}   1. Remove the old binary: rm ${path_binary}"
         echo -e "${YELLOW}!${NC}   2. Or adjust your PATH to prioritize ${install_dir}"
         echo ""
+        error "Installation completed but PATH isn't correct. Please check then restart the installation."
```
[the remark this answers is not in the document]
Comment (toothbrush) [reply scripts/install.sh]: Sure, this is technically correct, but we're doing this for a few reasons:

* We want to clarify to the user that things aren't "done" or "working" yet -> error state with a clear actionable instruction.
* We don't want to run postinstall yet, because it'll break your shell if our CLI isn't on PATH.  Let's get the user to fix that first before moving on.

[the remark this answers is not in the document]
Comment (toothbrush) [reply scripts/install.sh]: OK, i've redone it - i think onboarding is a lot friendlier now.

File: cmd/entire/cli/setup.go [hunk 6]
```diff
@@ -735,13 +739,12 @@ func promptShellCompletion(w io.Writer) error {
 		return nil
 	}
 
-	// Append completion to rc file
```
[the remark this answers is not in the document]
Comment (toothbrush) [reply cmd/entire/cli/setup.go:738]: There's a comment explaining that it isn't an error if the user dismisses the form:

https://github.com/entireio/cli/blob/e0324212b66bf445a941ab70c83f31376e5a98d5/cmd/entire/cli/setup.go#L733-L736

Also, from my local testing, it doesn't look like stdin is captured or the form is broken when piping the script to `bash`.

File: scripts/install.sh [hunk 7]
```diff
@@ -202,55 +168,95 @@ main() {
...
+                # shellcheck disable=SC2088
+                shell_config="~/.zshrc" ;;
+            bash)
+                if [[ -f "$HOME/.bash_profile" ]]; then
+                    # shellcheck disable=SC2088
+                    shell_config="~/.bash_profile"
+                else
+                    # shellcheck disable=SC2088
+                    shell_config="~/.bashrc"
+                fi
```
[the remark this answers is not in the document]
Comment (toothbrush) [reply scripts/install.sh:227]: I'm not convinced this is a really important discrepancy, nonetheless, addressed in https://github.com/entireio/cli/pull/180/commits/0bc8ed2ddcd406d047c8069d56158c7c3389edf3
